Makefile

```diff
@@ -0,0 +1,170 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# ==============================================================================
+# Build tunables
+
+# When updating these defaults, be sure to check that ALL_BUILDABLE_FLAVORS is updated
+FLAVORS_TO_BUILD ?= \
+	luminous,amd64,ubuntu,16.04,_,ubuntu,16.04 \
+	jewel,amd64,ubuntu,16.04,_,ubuntu,16.04 \
+	jewel,amd64,ubuntu,14.04,_,ubuntu,14.04 \
+	kraken,amd64,ubuntu,16.04,_,ubuntu,16.04 \
+
+REGISTRY ?= ceph
+
+
+# ==============================================================================
+# Internal definitions
+include maint-lib/makelib.mk
+
+# All flavor options that can be passed to FLAVORS_TO_BUILD
+ALL_BUILDABLE_FLAVORS := \
+	luminous,amd64,ubuntu,16.04,_,ubuntu,16.04 \
+	jewel,amd64,ubuntu,16.04,_,ubuntu,16.04 \
+	jewel,amd64,ubuntu,14.04,_,ubuntu,14.04 \
+	kraken,amd64,ubuntu,16.04,_,ubuntu,16.04 \
+
+
+# ==============================================================================
+# Build targets
+.PHONY: all stage build build.parallel build.all push
+
+stage.%:
+	@$(call set_env_var,CEPH_VERSION,$*) $(call set_env_var,ARCH,$*) \
+	$(call set_env_var,OS_NAME,$*) $(call set_env_var,OS_VERSION,$*) \
+	$(call set_env_var,BASEOS_REG,$*) $(call set_env_var,BASEOS_REPO,$*) \
+	$(call set_env_var,BASEOS_TAG,$*) $(call set_env_var,IMAGES_TO_BUILD,$*) \
+	$(call set_env_var,STAGING_DIR,$*) $(call set_env_var,BASE_IMAGE,$*) \
+	$(call set_env_var,DAEMON_BASE_IMAGE,$*) $(call set_env_var,DAEMON_IMAGE,$*) \
+	sh -c maint-lib/stage.py
+
+daemon-base.%: stage.%
+	@$(call set_env_var,STAGING_DIR,$*) ; $(MAKE) -C $$STAGING_DIR/daemon-base $(MAKECMDGOALS) \
+	$(call set_env_var,DAEMON_BASE_IMAGE,$*)
+
+daemon.%: daemon-base.%
+	@$(call set_env_var,STAGING_DIR,$*) ; $(MAKE) -C $$STAGING_DIR/daemon $(MAKECMDGOALS) \
+	$(call set_env_var,DAEMON_IMAGE,$*)
+
+do.image.%: daemon.% ;
+
+stage: $(foreach p, $(FLAVORS_TO_BUILD), stage.$(p)) ;
+build: $(foreach p, $(FLAVORS_TO_BUILD), do.image.$(p)) ;
+push:  $(foreach p, $(FLAVORS_TO_BUILD), do.image.$(p)) ;
+
+build.parallel:
+# Due to output-sync, will not output results until finished so there is no text interleaving
+	@$(MAKE) --jobs --output-sync build
+
+build.all:
+	@$(MAKE) FLAVORS_TO_BUILD="$(ALL_BUILDABLE_FLAVORS)" build.parallel
+
+
+# ==============================================================================
+# Clean targets
+.PHONY: clean clean.nones clean.all clean.nuke
+
+clean.image.%: do.image.%
```

Comment:
That's pretty curious to have do.image as a dependency of clean.image. That's explain why a make clean does populate the staging dir.

Btw, why not having clean.all being the default clean target ?
That looks like more consistent with a user expectation.